File: Python/lab6-1.py
def greedy_activity_selector(start_time, finish_time):
    n = len(start_time)
    selected_activities = []

    # The first activity is always selected
    selected_activities.append(0)

    # Consider the rest of the activities
    j = 0
    for i in range(1, n):
        # If this activity has a start time greater than or equal to the
        # finish time of the previously selected activity, then select it
        if start_time[i] >= finish_time[j]:
            selected_activities.append(i)
            j = i

    return selected_activities


# Example usage
start_times = [1, 3, 0, 5, 8, 5]
finish_times = [2, 4, 6, 7, 9, 9]

result = greedy_activity_selector(start_times, finish_times)
print("Selected activities:", result)

File: C/lab6-1.c
#include <stdio.h>

void greedy_activity_selector(int start_time[], int finish_time[], int n) {
    int arr[n];
    int count=1;
    int j=0;
    int i;
    arr[0] = 1;

    printf("Selected Activity : \n");

    for (i = 1; i < n; i++) {
        if (start_time[i] >= finish_time[j]) {
            arr[count] = i+1;
            count++;
            j = i;
        }
    }
    
    for(int i=0;i<count;i++){
        printf("-------");
    }
    printf("\n");
    for(int i=0 ; i<count ; i++){
        printf("  P%d  |", arr[i]);
    }
    printf("\n");
    for(int i=0;i<count;i++){
        printf("-------");
    }
    printf("\n");


}

int main() {
    int start_times[] = {1, 3, 0, 5, 8, 5,11,15};
    int finish_times[] = {2, 4, 6, 7, 9, 9,14,18};
    int n = sizeof(start_times) / sizeof(start_times[0]);

    greedy_activity_selector(start_times, finish_times, n);

    return 0;
}
